### PR TITLE
fix(query): show query with db not exists should return err

### LINKED ISSUE #8732: bug: show table from <db> return empty instead of report error when <db> not exists.
**Summary**

mysql report "Unknown database"

```
mysql> show tables from xxx;
ERROR 1049 (42000): Unknown database 'xxx'
```

cc @Chasen-Zhang

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Now:

```sql
show tables from `not_exists db` -- will return err

show table status from `not_exists db` -- will return err
```

Closes #8732

### CODE REVIEW

File: src/query/sql/src/planner/binder/ddl/table.rs [hunk 1]
```diff
@@ -166,10 +167,7 @@ impl<'a> Binder {
             with_history,
         } = stmt;
 
-        let database = database
-            .as_ref()
-            .map(|ident| normalize_identifier(ident, &self.name_resolution_ctx).name)
-            .unwrap_or_else(|| self.ctx.get_current_database());
+        let database = self.check_database_exist(database).await?;
```
Comment (Xuanwo) [inline src/query/sql/src/planner/binder/ddl/table.rs:170]: Nice change!

File: src/query/sql/src/planner/binder/ddl/table.rs [hunk 2]
```diff
@@ -340,6 +332,32 @@ impl<'a> Binder {
...
 
+    async fn check_database_exist(&mut self, database: &Option<Identifier<'_>>) -> Result<String> {
+        match database {
+            None => Ok(self.ctx.get_current_database()),
+            Some(ident) => {
+                let database = normalize_identifier(ident, &self.name_resolution_ctx).name;
+                let dbs = self
+                    .ctx
+                    .get_catalog(&self.ctx.get_current_catalog())?
+                    .list_databases(&self.ctx.get_tenant())
```
Comment (Xuanwo) [inline src/query/sql/src/planner/binder/ddl/table.rs:343]: Don't we have a `get_database` API? List all databases to check if it exists seems a bad pattern.

**Review by Xuanwo — DISMISSED** [review]
Nice refactor

**Review by Xuanwo — APPROVED** [review]

**Review by leiysky — APPROVED** [review]

**Comment by Xuanwo** [comment]
```shell
--- /workspace/tests/suites/0_stateless/13_tpch/13_0008_q8.result	2022-12-01 03:21:49.139685526 +0000
+++ /workspace/tests/suites/0_stateless/13_tpch/13_0008_q8.stdout	2022-12-01 03:25:45.317209375 +0000
@@ -1,2 +1 @@
-1995	0.02864874
-1996	0.01825027
+ERROR 1105 (HY000) at line 2: result row write failed: Code: 1067, displayText = error reading a body from connection: broken pipe, source: Some(hyper::Error(Body, Error { kind: Io(Kind(BrokenPipe)) })).
```

Not related. Let's merge.

**Pull request merged**